src/frequenz/sdk/timeseries/data_pipeline.py

```diff
@@ -0,0 +1,115 @@
+
+        Args:
+            resampler_config: Config to pass on to the resampler.
+            ev_charger_ids: Optional set of IDs of EV Charger to be managed by this
+                instance of the DataPipeline.
+        """
+        self._resampler_config = resampler_config
+        self._ev_charger_ids = ev_charger_ids
+
+        self._channel_registry = ChannelRegistry(name="Data Pipeline Registry")
+
+        self._data_sourcing_actor: _ActorInfo | None = None
+        self._resampling_actor: _ActorInfo | None = None
+        self._logical_meter: LogicalMeter | None = None
+        self._ev_charger_pool: EVChargerPool | None = None
+
+    def _data_sourcing_request_sender(self) -> Sender[ComponentMetricRequest]:
+        if self._data_sourcing_actor is None:
+            channel = Broadcast[ComponentMetricRequest](
+                "Data Pipeline: Data Sourcing Actor Request Channel"
+            )
+            actor = DataSourcingActor(
+                request_receiver=channel.new_receiver(), registry=self._channel_registry
+            )
+            self._data_sourcing_actor = _ActorInfo(actor, channel)
+        return self._data_sourcing_actor.channel.new_sender()
+
+    def _resampling_request_sender(self) -> Sender[ComponentMetricRequest]:
+        if self._resampling_actor is None:
+            channel = Broadcast[ComponentMetricRequest](
+                "Data Pipeline: Component Metric Resampling Actor Request Channel"
+            )
+            actor = ComponentMetricsResamplingActor(
+                channel_registry=self._channel_registry,
+                data_sourcing_request_sender=self._data_sourcing_request_sender(),
+                resampling_request_receiver=channel.new_receiver(),
+                config=self._resampler_config,
+            )
+            self._resampling_actor = _ActorInfo(actor, channel)
+        return self._resampling_actor.channel.new_sender()
+
+    def logical_meter(self) -> LogicalMeter:
+        """Return a reference to the logical meter instance.
+
+        Returns:
+            A logical meter instance.
+        """
+        if self._logical_meter is None:
+            self._logical_meter = LogicalMeter(
+                channel_registry=self._channel_registry,
+                resampler_subscription_sender=self._resampling_request_sender(),
+                component_graph=microgrid.get().component_graph,
+            )
+        return self._logical_meter
+
+    def ev_charger_pool(self) -> EVChargerPool:
+        """Return a reference to the EVChargerPool instance.
+
+        Returns:
+            An EVChargerPool instance.
+        """
+        if self._ev_charger_pool is None:
+            self._ev_charger_pool = EVChargerPool(
+                channel_registry=self._channel_registry,
+                resampler_subscription_sender=self._resampling_request_sender(),
+                component_ids=self._ev_charger_ids,
+            )
+        return self._ev_charger_pool
```

Comment:
We might need only one instance of ChannelRegistry and many instances of EvChargerPool.
Consider to add ev_charger_ids argument to this methods (instead of in constructor): 
```
def ev_charger_pool(self, ev_charger_ids: Set[int]) -> EVChargerPool:
```
And save them by components ids.
So if user will ask for ev_pool:
* with components ids that already was created, then you should return this EvPool.
* If EVPool with this ids does not exist, then class should create this EVPool, save and return.

Reply:
done